pkg/core/proxy/util/util.go

```diff
@@ -234,11 +235,29 @@ func ReadRequiredBytes(ctx context.Context, logger *zap.Logger, reader io.Reader
 // PassThrough function is used to pass the network traffic to the destination connection.
 // It also closes the destination connection if the function returns an error.
 func PassThrough(ctx context.Context, logger *zap.Logger, clientConn net.Conn, dstCfg *integrations.ConditionalDstCfg, requestBuffer [][]byte) ([]byte, error) {
+	logger.Debug("passing through the network traffic to the destination server", zap.Any("Destination Addr", dstCfg.Addr))
 	// making destConn
-	destConn, err := net.Dial("tcp", dstCfg.Addr)
-	if err != nil {
-		utils.LogError(logger, err, "failed to dial the destination server")
-		return nil, err
+	var destConn net.Conn
+	var err error
+	if dstCfg.TLSCfg != nil {
+		logger.Debug("trying to establish a TLS connection with the destination server", zap.Any("Destination Addr", dstCfg.Addr))
+		dialer := &net.Dialer{
```

Comment:
@gouravkrosx why do you have the timeout only for the tls conn?

Reply:
we can remove this timeout as the default timeout is already around 3 minutes.